packages/schematics/angular/utility/dependency_spec.ts

```diff
@@ -0,0 +1,263 @@
+        {
+          dependencies: { '@angular/common': '^14.0.0', '@angular/router': '^14.0.0' },
+        },
+        null,
+        2,
+      ),
+    );
+
+    const rule = addDependency('@angular/core', '^14.0.0');
+
+    await testRule(rule, tree);
+
+    expect(tree.readText('/package.json')).toBe(
+      JSON.stringify(
+        {
+          dependencies: {
+            '@angular/common': '^14.0.0',
+            '@angular/core': '^14.0.0',
+            '@angular/router': '^14.0.0',
+          },
+        },
+        null,
+        2,
+      ),
+    );
+  });
+
+  it('adds a dependency section if not present', async () => {
+    const tree = new EmptyTree();
+    tree.create('/package.json', JSON.stringify({}));
+
+    const rule = addDependency('@angular/core', '^14.0.0');
+
+    await testRule(rule, tree);
+
+    expect(tree.readJson('/package.json')).toEqual({
+      dependencies: { '@angular/core': '^14.0.0' },
+    });
+  });
+
+  it('adds a package to "devDependencies" when "type" is "dev"', async () => {
+    const tree = new EmptyTree();
+    tree.create(
+      '/package.json',
+      JSON.stringify({
+        dependencies: {},
+        devDependencies: {},
+      }),
+    );
+
+    const rule = addDependency('@angular/core', '^14.0.0', { type: 'dev' });
+
+    await testRule(rule, tree);
+
+    expect(tree.readJson('/package.json')).toEqual({
+      dependencies: {},
+      devDependencies: { '@angular/core': '^14.0.0' },
+    });
+  });
+
+  it('adds a package to "peerDependencies" when "type" is "peer"', async () => {
+    const tree = new EmptyTree();
+    tree.create(
+      '/package.json',
+      JSON.stringify({
+        devDependencies: {},
+        peerDependencies: {},
+      }),
+    );
+
+    const rule = addDependency('@angular/core', '^14.0.0', { type: 'peer' });
+
+    await testRule(rule, tree);
+
+    expect(tree.readJson('/package.json')).toEqual({
+      devDependencies: {},
+      peerDependencies: { '@angular/core': '^14.0.0' },
+    });
+  });
+
+  it('uses specified manifest when provided via "manifestPath" option', async () => {
+    const tree = new EmptyTree();
+    tree.create('/package.json', JSON.stringify({}));
+    tree.create('/abc/package.json', JSON.stringify({}));
+
+    const rule = addDependency('@angular/core', '^14.0.0', { manifestPath: '/abc/package.json' });
+
+    await testRule(rule, tree);
+
+    expect(tree.readJson('/package.json')).toEqual({});
+    expect(tree.readJson('/abc/package.json')).toEqual({
+      dependencies: { '@angular/core': '^14.0.0' },
+    });
+  });
+
+  it('schedules a package install task', async () => {
+    const tree = new EmptyTree();
+    tree.create('/package.json', JSON.stringify({}));
+
+    const rule = addDependency('@angular/core', '^14.0.0');
+
+    const tasks = await testRule(rule, tree);
+
+    // Should be an array with a single element with workingDirectory set
+    expect(tasks).toEqual([jasmine.objectContaining({ workingDirectory: '/' })]);
```

Comment:
Is there any indication that this is the `NodePackageInstallTask`? Could we do an `instanceof` check to make sure the right task got scheduled? As is, it's hard to be sure that `workingDirectory` actually means we made the right task.